src/client/components/BackButton/BackButton.tsx

```diff
@@ -24,9 +24,8 @@ export const BackButton = ({ onClick }: Props) => {
   const history = useHistory()
 
   const handleClick = () => {
-    history.goBack()
-
-    onClick?.()
+    if (onClick !== undefined && typeof onClick === 'function') onClick()
+    else history.goBack()
```

Comment:
Wouldn't it be enough to do?

```suggestion
    if (onClick) return onClick()
    history.goBack()
```